### PR TITLE
chore: Expose pending and rate metrics in 1/5/15 mins

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

Expose pending messages and processing rate in the past 1/5/15 minutes which can be used for UI display, and `default` for auto scaling (based on configurable lookback seconds)

```
# HELP vertex_pending_messages Average pending messages in the last period of seconds. It is the pending messages of a vertex, not a pod.
# TYPE vertex_pending_messages gauge
vertex_pending_messages{period="15m",pipeline="simple-pipeline",vertex="p1"} 0
vertex_pending_messages{period="1m",pipeline="simple-pipeline",vertex="p1"} 0
vertex_pending_messages{period="5m",pipeline="simple-pipeline",vertex="p1"} 0
vertex_pending_messages{period="default",pipeline="simple-pipeline",vertex="p1"} 0
# HELP vertex_processing_rate Message processing rate in the last period of seconds, tps. It represents the rate of a vertex instead of a pod.
# TYPE vertex_processing_rate gauge
vertex_processing_rate{period="15m",pipeline="simple-pipeline",vertex="p1"} 23.47826086956522
vertex_processing_rate{period="1m",pipeline="simple-pipeline",vertex="p1"} 25.714285714285715
vertex_processing_rate{period="5m",pipeline="simple-pipeline",vertex="p1"} 23.47826086956522
vertex_processing_rate{period="default",pipeline="simple-pipeline",vertex="p1"} 23.47826086956522
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,7 +31,8 @@ ARG ARCH
 COPY --from=base /usr/share/zoneinfo /usr/share/zoneinfo
 COPY --from=base /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 COPY --from=base /bin/numaflow /bin/numaflow
-COPY --from=numaflow-ui /app/ui/build /ui/build
+# Disable it as it is too slow
+#COPY --from=numaflow-ui /app/ui/build /ui/build
```
Comment (whynowy) [inline Dockerfile:35]: This is too slow, disable it temporarily.

**Review by vigith — APPROVED** [review]

File: pkg/isb/jetstream/reader.go [hunk 2]
```diff
@@ -160,12 +160,12 @@ func (jr *jetStreamReader) Rate(_ context.Context) (float64, error) {
 	startSeqInfo := timestampedSeqs[len(timestampedSeqs)-2]
 	for i := len(timestampedSeqs) - 3; i >= 0; i-- {
 		startSeqInfo = timestampedSeqs[i]
-		if endSeqInfo.timestamp-timestampedSeqs[i].timestamp > jr.opts.rateLookbackSeconds {
```
[the remark this answers is not in the document]
Comment (whynowy) [reply pkg/isb/jetstream/reader.go:163]: We still have `lookbackSeconds` configuration, which is set in upper layer. Here we can make it possible to query with different parameters.

File: pkg/isb/jetstream/reader.go [hunk 3]
```diff
@@ -160,12 +160,12 @@ func (jr *jetStreamReader) Rate(_ context.Context) (float64, error) {
...
+		if endSeqInfo.timestamp-timestampedSeqs[i].timestamp > seconds {
 			break
 		}
 	}
 	// Check if it is too stale (use lookbackSeconds + 2 * interval to determine)
```
[the remark this answers is not in the document]
Comment (whynowy) [reply pkg/isb/jetstream/reader.go]: Removed that check, we don't need it any more.

File: pkg/isb/jetstream/writer.go [hunk 4]
```diff
@@ -159,12 +159,12 @@ func (jw *jetStreamWriter) Rate(_ context.Context) (float64, error) {
 	startSeqInfo := timestampedSeqs[len(timestampedSeqs)-2]
 	for i := len(timestampedSeqs) - 3; i >= 0; i-- {
 		startSeqInfo = timestampedSeqs[i]
-		if endSeqInfo.timestamp-timestampedSeqs[i].timestamp > jw.opts.rateLookbackSeconds {
+		if endSeqInfo.timestamp-timestampedSeqs[i].timestamp > seconds {
 			break
 		}
 	}
 	// Check if it is too stale (use lookbackSeconds + 4 * interval to determine)
```
[the remark this answers is not in the document]
Comment (whynowy) [reply pkg/isb/jetstream/writer.go]: Same here, removed.

**Pull request merged**